### ISSUE TITLE
Feature request: Add support for additional specific drivers (i.e - delegations)

### ISSUE THREAD

**Issue by NirGallner** [bonigarcia/selenium-jupiter#67]
**Description of the problem**: Currently, selenium-jupiter supports a predefined list of browsers, stated in the `SeleniumExtension.class` constructor, or a general generic handler in `OtherDriverHandler.class` . I am using a `DelegateWebDriver.class`:
```
public class DelegateWebDriver implements WebDriver{
    public DelegateWebDriver(WebDriver driver) {
		super(driver);
    }
 
 // Some delegation code

}
```
The problems are:
1. `OtherDriverHandler.class` only handles general drivers which must implement a parameter-free c-tor.
2. There are specific initializations I would like to add in my `DelegateWebDriver.class`, and cannot with `OtherDriverHandler.class` 

**Proposed usage example (see my proposed solution after this)**:  I would like to extend `SeleniumExtension.java`  and `DriverHandler.java`  in order to do the following:

in `myTest.java` :
```
@ExtendWith(MySeleniumExtension.class)
public class JunitExampleTest {

    @Test
    public void test1(DelegateWebDriver driver) {
        driver.get("http://www.google.com");
    }
}
```

in `MySeleniumExtension.java` :
```
public class MySeleniumExtension extends SeleniumExtension {
    public MySeleniumExtension() {
        super();
        this.addEntry(this.handlerMap, "package.of.class.DelegateWebDriver", DelegateWebDriverHandler.class);
        this.addEntry(this.templateHandlerMap, "DelegateWebDriver", DelegateWebDriver.class);
    }
}
```

lastly,  in `DelegateWebDriverHandler.java` :
```
public class DelegateWebDriverHandler extends DriverHandler {
    public DelegateWebDriverHandler(Config config, AnnotationsReader annotationsReader) {
        super(config, annotationsReader);
    }

    public DelegateWebDriverHandler(Parameter parameter, ExtensionContext context, Config config, AnnotationsReader annotationsReader) {
        super(parameter, context, config, annotationsReader);
    }

    public void resolve() {
        try {
            Optional<Object> testInstance = this.context.getTestInstance();

           // initialization code goes here
            this.object = new DelegateWebDriver(...);

        } catch (Exception var4) {
            this.handleException(var4);
        }
    }
}
```

**Proposed solution**:
perform the following changes in the following files, **in order to allow extending `SeleniumExtension.java` and add custom handlers** (all changes are basically changing modifiers to `protected`:

1. In `SeleniumExtension.java` -
A. Make the constructor `protected` :
instead of:
```
SeleniumExtension() { ... }
```
define::
```
protected SeleniumExtension() { ... }
```

B. instead of: 
`private Map<String, Class<?>> handlerMap` 
define:
`protected Map<String, Class<?>> handlerMap`
(or create a relevant method to add to `handlerMap`)

C. instead of: 
`private Map<String, Class<?>> templateHandlerMap` 
define:
`protected Map<String, Class<?>> templateHandlerMap`
(or create a relevant method to add to `templateHandlerMap`)

D. instead of:
`private void addEntry(...)`
define:
`protected void addEntry`

2. In `DriverHandler.java` 
A. instead of:
```
Config config;
AnnotationsReader annotationsReader;
Parameter parameter;
ExtensionContext context;
Map<String, DockerContainer> containerMap;
DockerService dockerService;
Object object;
```

define:

   ```
protected Config config; 
protected AnnotationsReader annotationsReader;
protected Parameter parameter;
protected ExtensionContext context;
protected Map<String, DockerContainer> containerMap;
protected DockerService dockerService;
 protected Object object;
```
B. Instead of:
`void handleException(Exception e){...}`
define:
`protected void handleException(Exception e){...}`

C. instead of:
`File getExtension(String fileName) {...}`
define:
`protected File getExtension(String fileName) {...}`

**Selenium-Jupiter version**: 3.3.2

**Comment by bonigarcia** [comment]
@NirGallner Could you please prepare a PR about this?

**Comment by NirGallner** [comment]
Done. PR #68

**Comment by NirGallner** [comment]
Here is a draft suggestion to add to Selenium-Jupiter documentation:

**Custom Driver**
As of version XX.XX Selenium-Jupiter also supports custom drivers. Custom drivers support was added in order to:
a. support unknown driver classes (e.g delegations or any type of class which implements WebDriver); 
b. seamlessly support new types of browsers.

The following example shows how a delegate WebDriver class can be added to and supported by Selenium-Jupiter:

**Custom driver class**
Consider the following new WebDriver implementation:
```
public class DelegateWebDriver implements WebDriver{
    public DelegateWebDriver(WebDriver driver) {
		super(driver);
    }
 
 // Some delegation code and initializtion

}
```
We will need to define a new handler which will create an instance of `DelegateWebDriver within Selenium-Jupiter`:
```
public class DelegateWebDriverHandler extends DriverHandler {
    public DelegateWebDriverHandler(Config config, AnnotationsReader annotationsReader) {
        super(config, annotationsReader);
    }

    public DelegateWebDriverHandler(Parameter parameter, ExtensionContext context, Config config, AnnotationsReader annotationsReader) {
        super(parameter, context, config, annotationsReader);
    }

    public void resolve() {
        try {
            Optional<Object> testInstance = this.context.getTestInstance();

           // initialization code goes here
            this.object = new DelegateWebDriver(...);

        } catch (Exception var4) {
            this.handleException(var4);
        }
    }
}
```
Once we have our driver and our handler, we need to register the new `DelegateWebDriver `and its handler as an additional driver in a custom made extension:
```
public class MySeleniumExtension extends SeleniumExtension {
    public MySeleniumExtension() {
        super();
        this.addEntry(this.handlerMap, "package.of.class.DelegateWebDriver", DelegateWebDriverHandler.class);
        this.addEntry(this.templateHandlerMap, "DelegateWebDriver", DelegateWebDriver.class);
    }
}
```
and then we can use `DelegateWebDriver` in test instances:
```
@ExtendWith(MySeleniumExtension.class)
public class JunitExampleTest {

    @Test
    public void test1(DelegateWebDriver driver) {
        driver.get("http://www.google.com");
    }
}
```

**Comment by bonigarcia** [comment]
Thanks a lot for your contribution. It is already available in version 3.3.3. The doc has been also updated.
